### PR TITLE
add `configs` to validate_arguments

### PR DESCRIPTION
**Pull request by xbanke**
## Change Summary

This adds a kwargs `configs` to `validate_arguments` which can be used to set model config. It's just a sample code without test, but a reference.

## Related issue number
#1376

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: pydantic/decorator.py [hunk 1]
```diff
@@ -14,20 +14,29 @@
     Callable = TypeVar('Callable', bound=AnyCallable)
 
 
-def validate_arguments(function: 'Callable') -> 'Callable':
+def _validate_arguments(**configs):
```
Comment (samuelcolvin) [inline pydantic/decorator.py:17]: I don't think this needs to be a separate function.

File: pydantic/decorator.py [hunk 2]
```diff
@@ -14,20 +14,29 @@
...
+            return vd.call(*args, **kwargs)
+
+        wrapper_function.vd = vd  # type: ignore
+        wrapper_function.raw_function = vd.raw_function  # type: ignore
+        wrapper_function.model = vd.model  # type: ignore
+        return cast('Callable', wrapper_function)
+    return validate
+
+
+def validate_arguments(func: 'Callable' = None, **configs):
```
Comment (samuelcolvin) [inline pydantic/decorator.py:32]: what if someone wants to pass a `Config` class?

File: pydantic/decorator.py [hunk 3]
```diff
@@ -100,7 +109,7 @@ def __init__(self, function: 'Callable'):
             # same with kwargs
             fields[self.v_kwargs_name] = Dict[Any, Any], None
 
-        self.create_model(fields, takes_args, takes_kwargs)
+        self.create_model(fields, takes_args, takes_kwargs, **configs)
```
Comment (samuelcolvin) [inline pydantic/decorator.py:112]: better in cases like this to pass the dict, not kwargs kwargs requires a shallow copy of the dict.

**Comment by samuelcolvin** [comment]
@xbanke any change you could finish this?

**Comment by xbanke** [comment]
@samuelcolvin , sorry for replying too late. I merged the functions. I am thinking about adding a dict parameter, mybe named aliases, to solve name conflict.

File: pydantic/decorator.py (hunk 3 above)
Comment (xbanke) [reply pydantic/decorator.py:120]: Maybe we should add a parameter `model` that users can use cumtom model class.

Comment (xbanke) [reply pydantic/decorator.py:120]: Because `extra` should always be `forbid`, so there still exists copy of dict if pass a dict.

**Comment by StephenBrown2** [comment]
You'll need to change all the double quotes, as this repository prefers single quotes for everything.

**Comment by xbanke** [comment]
> You'll need to change all the double quotes, as this repository prefers single quotes for everything.

Sorry, that maybe caused by `black`, I've repaired that.

**Comment by StephenBrown2** [comment]
Another lint! This time it looks like a different line width was used, probably also due to black's default 88 line width:
```diff
 ERROR: /home/runner/work/pydantic/pydantic/pydantic/decorator.py Imports are incorrectly sorted.
--- /home/runner/work/pydantic/pydantic/pydantic/decorator.py:before	2020-05-21 00:44:29.613869
+++ /home/runner/work/pydantic/pydantic/pydantic/decorator.py:after	2020-05-21 00:45:13.522116
@@ -1,15 +1,5 @@
 from functools import wraps
-from typing import (
-    TYPE_CHECKING,
-    Any,
-    Dict,
-    List,
-    Mapping,
-    Tuple,
-    TypeVar,
-    cast,
-    get_type_hints,
-)
+from typing import TYPE_CHECKING, Any, Dict, List, Mapping, Tuple, TypeVar, cast, get_type_hints
 
```

Just a heads up, you can use `make lint` in the root of the repo to check the same thing that CI is checking, and `make format` to fix things up as it expects.

**Comment by xbanke** [comment]
@StephenBrown2 , thanks for your patience, I've fixed that.

**Comment by samuelcolvin** [comment]
This is still failing.

**Comment by xbanke** [comment]
I fixed some type hints and `make mypy` is passed, but still errors with mypy: `maximum semantic analysis iteration count reached`.

**Comment by samuelcolvin** [comment]
Thank so much.

I've just realised we forgot to add documentation or tests for this. My fault for merging to quickly, @xbanke any chance you could do that? Otherwise I'll submit a PR for it.

**Comment by xbanke** [comment]
> Thank so much.
> 
> I've just realised we forgot to add documentation or tests for this. My fault for merging to quickly, @xbanke any chance you could do that? Otherwise I'll submit a PR for it.

Sorry for that, I am quite busy these days. Please submit a PR. @samuelcolvin

**Comment by samuelcolvin** [comment]
ok, I'm working on it.

**Pull request merged**